### ISSUE TITLE
xi2 License Dependency

### ISSUE THREAD

**Issue by cliveseldon** [mholt/archiver#259]
This project as xi2/xz as a dependency

https://github.com/mholt/archiver/blob/25e050dbb4e3c591402916201098241bb661a2f7/go.mod#L14

However, that project has not used a standard license:

https://github.com/xi2/xz/blob/master/LICENSE

This is causing us problems with 3rd party open source dependency license validation. Is there any possibility of forking that library to republish it with a standard license? I have added an issue to their project https://github.com/xi2/xz/issues/10 but it seems inactive.

**Comment by mholt** [comment]
I hadn't noticed... are you really not able to use software in the public domain?

**Comment by cliveseldon** [comment]
We are and our software is open source but unless a software project we use has a legally well known and clarified license it makes things less clear and harder to verify its status.

**Comment by coolaj86** [comment]
@mholt I think the problem is that "Public Domain" is not a license that can be detected with automated tooling.The MIT license et al can be automatically detected with tooling because they use exact phrases and wording.

**Comment by cliveseldon** [comment]
Another issue is marketplaces such as Google accept a limited set of well established licences. By taking the approach of creating a custom license with "reasonable sounding" words makes it harder for these software projects to be accepted when scrutinized in strict environments. Its a shame as I am sure the xi2 project would probably be happy with MIT or Apache 2.0 wording.

**Comment by coolaj86** [comment]
Having not heard anything back from the project's author, I created a fork with the LICENSE:

https://github.com/therootcompany/xz

Perhaps you could use the `replace` directive in `go.mod`?

**Comment by mholt** [comment]
Fixed in #302.

Tried using a `replace` directive but got this error:

```
go: finding module for package github.com/therootcompany/xz
go: found github.com/therootcompany/xz in github.com/therootcompany/xz v1.0.1
go: github.com/therootcompany/xz@v1.0.1 used for two different module paths (github.com/therootcompany/xz and github.com/xi2/xz)
```

so I had to just swap it out entirely, nbd. Thanks!
